### ISSUE TITLE
FNX-5414 ⁃ [Bug] When TalkBack is disabled with Fenix running, the bottom bar does not scroll away

### ISSUE THREAD

**Issue by eeejay** [mozilla-mobile/fenix#2035]
## Steps to reproduce
1. Enable TalkBack
1. Start Fenix and go to a web site
1. Disable TalkBack
1. Return to Fenix

### Expected behavior
The bottom toolbar should be dynamic and scroll away when scrolling.

### Actual behavior
The bottom toolbar remains "pinned"

### More Info
The opposite scenario is also a bug: enabling TalkBack after Fenix is started makes the dynamic toolbar go way with TalkBack enabled.

This seems to be because the coordinator layout is instantiated at startup depending on the state of TalkBack, but there is no listener for when this changes mid app lifetime.

**Comment by sv-ohorvath** [comment]
I can still reproduce this issue on the 4/30 build. @eeejay Is this the same as your original issue?
Please note that in the first part of the gif, the toolbar is hiding while TalkBack is on. In the second part, the toolbar is pinned after turning TalkBack off.
![20190430_175619](https://user-images.githubusercontent.com/4037709/56971448-e7721280-6b71-11e9-88ec-188c1eee6458.gif)

**Comment by sv-ohorvath** [comment]
@ekager I'm still reproducing this on Pixel :(

**Comment by ekager** [comment]
Is it reproducing with the same STR? I don't have a TalkBack shortcut set up but I can't reproduce with: 
1. Load page with TB off (toolbar should scroll)
2. Turn on TB
3. Try to scroll page (toolbar should not scroll)

and the reverse. Maybe it's the TB shortcut not triggering onTouchExplorationStateChanged but I'm not sure

**Comment by eeejay** [comment]
Yup I'm still seeing this too. 
1. I see that @ekager made `BrowserFragment` a `TouchExplorationStateChangeListener`, but I don't see where it actually gets added.
1. github newb question. I see the various commit revisions inline here, how do I actually find the pull request?

**Comment by ekager** [comment]
Figured out the shortcut and have a fix 👍

**Comment by bsurd** [comment]
> Is it reproducing with the same STR? I don't have a TalkBack shortcut set up but I can't reproduce with:
> 
> 1. Load page with TB off (toolbar should scroll)
> 2. Turn on TB
> 3. Try to scroll page (toolbar should not scroll)

The issue is not reproducible anymore with the talkback option offered by Google. Tested with the latest build (5/15) on a Pixel, OnePlus 5T and HTC 10.

It still affects some devices (eg. Samsung) that have their own voice assistant, the URL bar hides with talkback enabled. I have logged #2508 for this.
